modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/ElasticSearchChecker.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     bdelbosc
+ */
+package org.nuxeo.elasticsearch;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.cluster.health.ClusterHealthStatus;
+import org.nuxeo.common.xmap.XMap;
+import org.nuxeo.elasticsearch.api.ESClient;
+import org.nuxeo.elasticsearch.api.ESClientFactory;
+import org.nuxeo.elasticsearch.client.ESRestClientFactory;
+import org.nuxeo.elasticsearch.config.ElasticSearchClientConfig;
+import org.nuxeo.launcher.config.ConfigurationException;
+import org.nuxeo.launcher.config.ConfigurationGenerator;
+import org.nuxeo.launcher.config.backingservices.BackingChecker;
+
+/**
+ * @since 11.3
+ */
+public class ElasticSearchChecker implements BackingChecker {
+    private static final Logger log = LogManager.getLogger(ElasticSearchChecker.class);
+
+    protected static final String ELASTIC_ENABLED_OPT = "elasticsearch.enabled";
+
+    protected static final String ELASTIC_REST_CLIENT_OPT = "elasticsearch.client";
+
+    protected static final String ADDRESS_LIST_OPT = "addressList";
+
+    protected static final String CONFIG_NAME = "elasticsearch-config.xml";
+
+    @Override
+    public boolean accepts(ConfigurationGenerator cg) {
+        if (!"true".equals(cg.getUserConfig().getProperty(ELASTIC_ENABLED_OPT))) {
```

Comment:
Actually `Boolean.valueOf` -> `Boolean.parseBoolean`

Reply:
Using `Boolean.parseBoolean` will trigger the checker for value like`True` or `TRUE`,
but in our template system, the feature will not be activated (only `true` is taken in account) 
this will lead to buggy behavior.

Reply:
Worth a comment though, so that others don't refactor the code later in the way we just suggested here...